### DISCUSSION TITLE
How to define dynamic variable

### DISCUSSION THREAD

**Discussion by nucleartux** [facebook/stylex#206] [Q&A] [answered]
How can I write variables that come dynamically, for example, from an API? I want to create a variable using defineVars, but I don't know how to do that.

```
const colors = stylex.defineVars({
  accent: '???',
});
```

Normally, I can set CSS variables inside the style tag in my application root.

**Comment by nmn** [comment, marked as answer]
# Step 1 

Define default values for variables using `defineVars`. These default values *cannot* be dynamic. These are the default values.

# Step 2

Use `stylex.create` to set values for variables dynamically:

```tsx
const styles = stylex.create({
  dynamicTheme: (themeFromAPI) => ({
    [colors.accent]: themeFromAPI.accent,
  }),
});

// Somewhere near the top of the app:
<html {...stylex.props(styles.dynamicTheme(themeFromAPI))}>
  {children}
</html>
```


## NOTE:

- This solution works and the API is stable.
- There will further performance optimisations for this in the near future.
- We might also add support for functions to `createTheme` if this is a common use-case
